### PR TITLE
LGTM deprecation: updates to CodeQL for Java articles

### PR DESCRIPTION
**Pull request by felicitymay**
This pull request updates the language guides for Java to remove references to LGTM.

As far as I can tell, most of the references can be removed without substitution. If I've misjudged this, please let me know.

This is one of many pull requests. I've raised a separate pull request to update the "Basic query" articles so that they guide users through get started running queries in VS Code. This should help offset the removal of the links to the query console: https://github.com/github/codeql/pull/11423.

### CODE REVIEW

File: docs/codeql/codeql-language-guides/codeql-library-for-java.rst [hunk 1]
```diff
@@ -278,7 +270,7 @@ For annotations, class ``Annotatable`` is a superclass of all program elements t
    from Constructor c
    select c.getAnAnnotation()
 
-➤ `See this in the query console on LGTM.com <https://lgtm.com/query/3206112561297137365/>`__. The LGTM.com demo projects all use annotations, you can see examples where they are used to suppress warnings and mark code as deprecated.
+You may see examples where annottions are used to suppress warnings or to mark code as deprecated.
```
Comment (atorralba) [inline docs/codeql/codeql-language-guides/codeql-library-for-java.rst:273]:
```suggestion
You may see examples where annotations are used to suppress warnings or to mark code as deprecated.
```

File: docs/codeql/codeql-language-guides/codeql-library-for-java.rst [hunk 2]
```diff
@@ -307,8 +299,6 @@ For Javadoc, class ``Element`` has a member predicate ``getDoc`` that returns a
        jdoc = f.getDoc().getJavadoc()
    select jdoc
 
-➤ `See this in the query console on LGTM.com <https://lgtm.com/query/6022769142134600659/>`__. You can see this pattern in many projects.
```
Comment (atorralba) [inline docs/codeql/codeql-language-guides/codeql-library-for-java.rst:310]: Is the "You can see this pattern in many projects." part worth keeping?

File: docs/codeql/codeql-language-guides/codeql-library-for-java.rst [hunk 3]
```diff
@@ -381,7 +367,7 @@ Conversely, ``Callable.getAReference`` returns a ``Call`` that refers to it. So
    where not exists(c.getAReference())
    select c
 
-➤ `See this in the query console on LGTM.com <https://lgtm.com/query/7261739919657747703/>`__. The LGTM.com demo projects all appear to have many methods that are not called directly, but this is unlikely to be the whole story. To explore this area further, see ":doc:`Navigating the call graph <navigating-the-call-graph>`."
+Codebases often have many methods that are not called direcstly, but this is unlikely to be the whole story. To explore this area further, see ":doc:`Navigating the call graph <navigating-the-call-graph>`."
```
Comment (atorralba) [inline docs/codeql/codeql-language-guides/codeql-library-for-java.rst:370]:
```suggestion
Codebases often have many methods that are not called directly, but this is unlikely to be the whole story. To explore this area further, see ":doc:`Navigating the call graph <navigating-the-call-graph>`."
```

File: docs/codeql/codeql-language-guides/types-in-java.rst [hunk 4]
```diff
@@ -80,8 +80,6 @@ This recipe is not too difficult to translate into a query:
        target.getElementType().(RefType).getASupertype+() = source.getElementType()
    select ce, "Potentially problematic array downcast."
 
-➤ `See this in the query console on LGTM.com <https://lgtm.com/query/8378564667548381869/>`__. Many projects return results for this query.
```
Comment (atorralba) [inline docs/codeql/codeql-language-guides/types-in-java.rst:83]: Is the "Many projects return results for this query." part worth keeping?

**Review by atorralba — COMMENTED** [review]
Mostly looks good to me, I added a few comments that might be worth addressing.

File: docs/codeql/codeql-language-guides/codeql-library-for-java.rst (hunk 2 above)
Comment (felicitymay) [reply docs/codeql/codeql-language-guides/codeql-library-for-java.rst:310]: Happy to keep it 👍🏻

**Comment by felicitymay** [comment]
@atorralba - many thanks for saving my blushes and picking up those typos 🤦🏻‍♀️ 

I've also made the other updates you suggested.

**Review by atorralba — DISMISSED** [review]
Thanks for addressing the comments!

**Review by hubwriter — APPROVED** [review]
All good 👍

**Comment by felicitymay** [comment]
Following the change of branch, I've lost my approving reviews. Would someone be able to give this another 👍🏻

**Review by atorralba — APPROVED** [review]

**Comment by felicitymay** [comment]
Many thanks @atorralba 💖

**Pull request merged**